File: 寻找数组的中心下标.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        all = sum(nums[1:])
        left = 0
        if left == all:
            return 0
        for i, x in enumerate(nums[1:], 1):
            left += nums[i - 1]
            all -= x
            if left == all:
                return i
        return -1

File: 寻找数组的中心下标.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n = nums.size();
        vector<int> pre(n, nums[0]);
        for (int i = 1; i < n; ++i) {
            pre[i] = pre[i - 1] + nums[i];
        }
        if (pre[n - 1] - pre[0] == 0) {
            return 0;
        }
        for (int i = 1; i < n; ++i) {
            if (pre[i - 1] == pre[n - 1] - pre[i]) {
                return i;
            }
        }
        return -1;
    }
};
